utils/mc_log_gui/mc_log_ui/mc_log_plotcanvas.py

```diff
@@ -1090,15 +1178,25 @@ def stopAnimation(self):
     self.draw()
 
   def saveAnimation(self):
-    fpath = QtWidgets.QFileDialog.getSaveFileName(self, "Output file", filter = "Video (*.mp4)")[0]
+    fpath = QtWidgets.QFileDialog.getSaveFileName(self,  "Output animation", self.animation_path+"/output-animation.mp4", filter = "Video (*.mp4)")[0]
     if not len(fpath):
       return
+    self.animation_path = os.path.split(fpath)[0]
+    filename, extension = os.path.splitext(fpath)
+    if not extension:
+      fpath = fpath + '.mp4'
+    print "Saving animation, please wait..."
```

Comment:
That would be nice to show a dialog here instead of a console message

Reply:
Yes I agree, saving anything realistic can take several minutes.
This seemed a bit tricky to do, and I wanted to move on to other things so I didn't look deeply into it. From what I saw, it seemed a bit tricky to know the saving progression, and currently the saving call is blocking. I guess at least we could display a spinning progress bar in a modal window until saving is finished. This would imply running the saving process in a thread, which might have other implications (should be fine though as long as the GUI cannot be used while the modal progress window is displayed).

I'll try looking into it, I have a few additional bugfixes to add to this PR, and Mehdi is intensively testing this PR for his upcoming paper.